src/highlightjs-adapter.ts

```diff
@@ -0,0 +1,36 @@
+// from https://github.com/cdnjs/SRIs/blob/master/highlight.js/9.18.3.json
+import * as sri from './highlightjs-sri-9.18.3.json'
+const {Opal} = require('asciidoctor-opal-runtime')
```

Comment:
Asciidoctor.js does not provide an API to extend a class, that's why we are using Opal directly.

We could introduce a third argument on the `SyntaxHighlighter.register` function:


```js
class CustomHighlightJsAdapter {
  docinfo (location) {
    // ...
  }
}

asciidoctor.SyntaxHighlighter.register(
  // names
  ['highlight.js', 'highlightjs'],
  // syntax highligther
  CustomHighlightJsAdapter,
  // extends
  asciidoctor.SyntaxHighlighter.for('highlight.js')
)
```

Or introduce a generic function:

```js
class CustomHighlightJsAdapter {
  docinfo (location) {
    // ...
  }
}

asciidoctor.klass.extend(CustomHighlightJsAdapter, asciidoctor.SyntaxHighlighter.for('highlight.js'))

asciidoctor.SyntaxHighlighter.register(['highlight.js', 'highlightjs'], CustomHighlightJsAdapter)
```

Reply:
I am so glad you are here to help :smiley_cat: